iodata/test/test_wfn.py

```diff
@@ -306,3 +308,88 @@ def test_load_one_cah110_hf_sto3g_g09():
     olp = compute_overlap(mol.obasis, mol.atcoords)
     check_orthonormal(mol.mo.coeffsa, olp, 1e-5)
     check_orthonormal(mol.mo.coeffsb, olp, 1e-5)
+
+
+def check_load_dump_consistency(fn, tmpdir, fmt_from='wfn', fmt_to='wfn'):
+    """Check if data is preserved after dumping and loading a WFN file.
+
+    Parameters
+    ----------
+    fn : str
+        The filename to load
+    tmpdir : str
+        The temporary directory to dump and load the file.
+    fmt_from : str
+        Format filename to load.
+    fmt_to : str
+        Format of filename to dump and then load again.
+
+    """
+    with path('iodata.test.data', fn) as file_name:
+        mol1 = load_one(str(file_name), fmt=fmt_from)
+    fn_tmp = os.path.join(tmpdir, 'foo.bar')
+    dump_one(mol1, fn_tmp, fmt=fmt_to)
+    mol2 = load_one(fn_tmp, fmt=fmt_to)
+    if fmt_from == fmt_to:
+        compare_mols(mol1, mol2, atol=1.0e-6)
+    else:
+        compare_mols_diff_formats(mol1, mol2)
+
+
+def test_load_dump_consistency_lih_cation_cisd(tmpdir):
+    check_load_dump_consistency('lih_cation_cisd.wfn', tmpdir)
+
+
+def test_load_dump_consistency_lih_cation_uhf(tmpdir):
+    check_load_dump_consistency('lih_cation_uhf.wfn', tmpdir)
+
+
+def test_load_dump_consistency_lih_cation_rohf(tmpdir):
+    check_load_dump_consistency('lih_cation_rohf.wfn', tmpdir)
+
+
+def test_load_dump_consistency_h2o(tmpdir):
+    check_load_dump_consistency('h2o_sto3g.wfn', tmpdir)
+    check_load_dump_consistency('h2o_sto3g_decontracted.wfn', tmpdir)
+
+
+def test_load_dump_consistency_lif(tmpdir):
+    check_load_dump_consistency('lif_fci.wfn', tmpdir)
```

Comment:
@msricher can you please check why this test fails? It has fractional orbital occupations.